cython/_pocketsphinx.pxd

```diff
@@ -0,0 +1,183 @@
+    int cmd_ln_free_r(cmd_ln_t *cmdln)
+
+    double cmd_ln_float_r(cmd_ln_t *cmdln, const char *name)
+    int cmd_ln_type_r(cmd_ln_t *cmdln, const char *name)
+    long cmd_ln_int_r(cmd_ln_t *cmdln, const char *name)
+    const char *cmd_ln_str_r(cmd_ln_t *cmdln, const char *name)
+    void cmd_ln_set_str_r(cmd_ln_t *cmdln, const char *name, const char *str)
+    void cmd_ln_set_str_extra_r(cmd_ln_t *cmdln, const char *name, const char *str)
+    void cmd_ln_set_int_r(cmd_ln_t *cmdln, const char *name, long iv)
+    void cmd_ln_set_float_r(cmd_ln_t *cmdln, const char *name, double fv)
+    cmd_ln_val_t *cmd_ln_access_r(cmd_ln_t *cmdln, const char *name)
+
+    int cmd_ln_exists_r(cmd_ln_t *cmdln, const char *name)
+
+
+cdef extern from "sphinxbase/fsg_model.h":
+    ctypedef struct fsg_model_t:
+        int start_state
+        int final_state
+
+    fsg_model_t *fsg_model_init(const char *name, logmath_t *lmath,
+                                float lw, int n_state)
+    fsg_model_t *fsg_model_readfile(const char *file, logmath_t *lmath,
+                                    float lw)
+    const char *fsg_model_name(fsg_model_t *fsg)
+    int fsg_model_free(fsg_model_t *fsg)
+
+    int fsg_model_word_add(fsg_model_t *fsg, const char *word)
+    int fsg_model_word_id(fsg_model_t *fsg, const char *word)
+    void fsg_model_trans_add(fsg_model_t * fsg,
+                             int source, int dest, int logp, int wid)
+    int fsg_model_null_trans_add(fsg_model_t * fsg, int source, int dest,
+                                 int logp)
+    int fsg_model_tag_trans_add(fsg_model_t * fsg, int source, int dest,
+                                int logp, int wid)
+    void fsg_model_writefile(fsg_model_t *fsg, const char *file)
+
+
+cdef extern from "sphinxbase/jsgf.h":
+    ctypedef struct jsgf_t:
+        pass
+    ctypedef struct jsgf_rule_t:
+        pass
+    fsg_model_t *jsgf_read_file(const char *name, logmath_t *lmath,
+                                float lw)
+    jsgf_t *jsgf_parse_string(const char *string, jsgf_t *parent)
+    const char *jsgf_grammar_name(jsgf_t *jsgf)
+    void jsgf_grammar_free(jsgf_t *jsgf)
+    jsgf_rule_t *jsgf_get_rule(jsgf_t *grammar, const char *name)
+    jsgf_rule_t *jsgf_get_public_rule(jsgf_t *grammar)
+    fsg_model_t *jsgf_build_fsg(jsgf_t *grammar, jsgf_rule_t *rule,
+                                logmath_t *lmath, float lw)
+
+
+cdef extern from "pocketsphinx.h":
+    ctypedef struct ps_decoder_t:
+        pass
+    ctypedef struct ps_seg_t:
+        pass
+    ctypedef struct ps_nbest_t:
+        pass
+    ctypedef struct ps_lattice_t:
+        pass
+    arg_t *ps_args()
+    ps_decoder_t *ps_init(cmd_ln_t *config)
+    int ps_free(ps_decoder_t *ps)
+    int ps_reinit(ps_decoder_t *ps, cmd_ln_t *config)
+    int ps_reinit_feat(ps_decoder_t *ps, cmd_ln_t *config)
+    logmath_t *ps_get_logmath(ps_decoder_t *ps)
+    int ps_start_stream(ps_decoder_t *ps)
+    int ps_get_in_speech(ps_decoder_t *ps)
```

Comment:
Can you expose get_in_speech as well? That would let you do some nice things like keep a mic open but discard the. audio up to x msec before speech starts

Reply:
ahh, funny you mention this, because I had removed get_in_speech as the internal VAD, which it used, was buggy and also not a very good VAD.  do you have code that uses it already? I am trying to figure out what exactly the use case is for it and whether it can be supported or not...

Reply:
No -- and actually we can just check if there are word in the hyp, so I wouldn't worry about it